src/methods.jl

```diff
@@ -156,15 +156,20 @@ function _check_p_q(p::Integer, q::Integer)
 end
 
 # Compute coefficients for the method
+
+const _COEFFS_CACHE = Dict{Tuple{AbstractVector{<:Real}, Integer, Integer}, Vector{Float64}}()
 function _coefs(grid::AbstractVector{<:Real}, p::Integer, q::Integer)
-    # For high precision on the \ we use Rational, and to prevent overflows we use Int128
-    # At the end we go to Float64 for fast floating point math (rather than rational math)
-    C = [Rational{Int128}(g^i) for i in 0:(p - 1), g in grid]
-    x = zeros(Rational{Int128}, p)
-    x[q + 1] = factorial(q)
-    return Float64.(C \ x)
+    return get!(_COEFFS_CACHE, (grid, p, q)) do
```

Comment:
Is it worth thinking about thread safety here?

Reply:
Could do, but  I don't think it is worth it.
There is much more thread unsafe stuff in this package that this.
(Mostly the history tracking stuff).